### DISCUSSION TITLE
Clips within mask, snaps outside

### DISCUSSION THREAD

**Discussion by Thunderbiscuits** [blakeblackshear/frigate#938] [Ask A Question]
Awesome work on this. Seeing it go through the different iterations has been really impressive!

I want to record videos of all people, animals, and cars that step onto my side of the curb.
I only want to record snapshots of people, animals, and cars that are on the road side of the curb.

Is this possible?
In my understanding, object masks apply to the detected object which is used for both clips and snapshots.

Can you apply filters to the objects defined in clips and snapshots for a specific camera?

thank you!

**Comment by blakeblackshear** [comment]
Masks should be used to filter areas where the object cannot possibly be and it is a guaranteed false positive. For example, masking the sky and roof for cars. Use masks sparingly since they hurt frigate's ability to track objects. 

To do what you want, define zones and use the required_zones option to ensure snapshots and clips are only created for the zones you want.
